sphinx/builders/__init__.py

```diff
@@ -257,38 +259,50 @@ def compile_update_catalogs(self) -> None:
 
     def build_all(self) -> None:
         """Build all source files."""
+        self.compile_all_catalogs()
```

Comment:
BTW, I think separate message catalog building process to individual class is better. It would works better if we'll do multiple builds at once running.

Reply:
As a trial, I made a example of above comment. With improving this, Sphinx might be able to build multiple output at once in future.
https://github.com/sphinx-doc/sphinx/compare/2.0...tk0miya:multiple_builders?expand=1

Reply:
> Why do you move these methods? I suppose these change is not related with "expand use of path2doc".

As noted [here](https://github.com/sphinx-doc/sphinx/pull/6945/commits/3eb6589637dab6fccbc4adc80bf3487700315353)

>  builder: Combine 'compile_xxx' and 'build_xxx' functions
> 
> These are only called from the one location and we already use 'build'
> to do both building and writing. Given that we intend to simplify the
> handling of the 'filenames' argument, this makes our lives there much
> easier.

It looks like there is good reason not to do this though, so I can revert if you want?

Reply:
I feel we need to design whole of builders and its interface carefully. I still don't know my rough code is good idea or not. So we should discuss this in another PR. +1 for reverting.